### PR TITLE
CLI core 100% covered + small refactoring.

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

CLI core 100% covered + small refactoring.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/core.py [hunk 1]
```diff
@@ -177,13 +187,20 @@ def add_key(ctx: Context, type_, file):
...
+        config_pair[0]: config_pair[1]
+        for config_pair in ctx.agent_config.private_key_paths.read_all()
+    }
     try:
-        ctx.agent_config.private_key_paths.create(type_, file)
+        wallet = Wallet(private_key_paths)
+        address = wallet.addresses[type_]
+        return address
     except ValueError as e:  # pragma: no cover
         logger.error(str(e))  # pragma: no cover
```
Comment (Totoual) [inline aea/cli/core.py]: You can remove the # pragma: no cover here :)

File: aea/cli/core.py [hunk 2]
```diff
@@ -177,13 +187,20 @@ def add_key(ctx: Context, type_, file):
     """Add a private key to the wallet."""
     try_to_load_agent_config(ctx)
     _validate_private_key_path(file, type_)
+    _add_key(ctx, type_, file)
+
+
+def _get_address(ctx, type_):
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: _try_get_address would be better

File: aea/cli/core.py [hunk 3]
```diff
@@ -177,13 +187,20 @@ def add_key(ctx: Context, type_, file):
...
-        ctx.agent_config.private_key_paths.create(type_, file)
+        wallet = Wallet(private_key_paths)
+        address = wallet.addresses[type_]
+        return address
     except ValueError as e:  # pragma: no cover
-        logger.error(str(e))  # pragma: no cover
-    ctx.agent_loader.dump(
-        ctx.agent_config, open(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w")
-    )
+        logger.error(str(e))
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: I think we should add a `sys.exit(1)` here

File: aea/cli/core.py [hunk 4]
```diff
@@ -229,6 +237,15 @@ def _try_get_balance(agent_config, wallet, type_):
         sys.exit(1)
 
 
+def _get_wealth(ctx, type_):
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: `_try_get_wealth`

File: aea/cli/core.py [hunk 5]
```diff
@@ -298,6 +294,24 @@ def generate_wealth(ctx: Context, sync, type_):
         logger.error(str(e))  # pragma: no cover
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: I think we should do `sys.exit(1)` here as well

File: aea/cli/core.py [hunk 6]
```diff
@@ -261,22 +272,7 @@ def _wait_funds_release(agent_config, wallet, type_):
...
-@click.option(
-    "--sync", is_flag=True, help="For waiting till the faucet has released the funds."
-)
-@pass_ctx
-def generate_wealth(ctx: Context, sync, type_):
-    """Generate wealth for address on test network."""
-    try_to_load_agent_config(ctx)
-
-    _verify_or_create_private_keys(ctx)
+def _generate_wealth(ctx, type_, sync):
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: `_try_generate_wealth`

File: aea/cli/core.py [hunk 7]
```diff
@@ -159,6 +159,16 @@ def _can_write(path) -> bool:
             EthereumCrypto().dump(open(ETHEREUM_PRIVATE_KEY_FILE, "wb"))
 
 
+def _add_key(ctx, type_, filepath):
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: `_try_add_key` is better

File: aea/cli/core.py [hunk 8]
```diff
@@ -159,6 +159,16 @@ def _can_write(path) -> bool:
...
+    try:
+        ctx.agent_config.private_key_paths.create(type_, filepath)
+    except ValueError as e:  # pragma: no cover
+        logger.error(str(e))  # pragma: no cover
```
Comment (DavidMinarsch) [inline aea/cli/core.py]: `sys.exit(1)` here please

**Pull request merged**